common/persistence/historyV2Store.go

```diff
@@ -291,6 +291,9 @@ func (m *historyV2ManagerImpl) readHistoryBranch(byBatch bool, request *ReadHist
 	}
 
 	resp, err := m.persistence.ReadHistoryBranch(req)
+	if len(resp.History) == 0 && len(request.NextPageToken) == 0 {
```

Comment:
If you get an error will resp be nil? i.e. can you get a nil pointer here?

Reply:
Updated.